framework/src/loops/UpdateDisplacedMeshThread.C

```diff
@@ -125,3 +133,9 @@ void
 UpdateDisplacedMeshThread::join(const UpdateDisplacedMeshThread & /*y*/)
 {
 }
+
+void
+UpdateDisplacedMeshThread::post()
+{
+  _ref_mesh.comm().max(_has_displacement);
```

Comment:
This has become a dubious optimization. Did you do any quantification of the number of times `UpdateDisplacedMeshThread` didn't trivially displace the mesh?

Reply:
I have cases where the displacement variables are set by aux kernels on initial or ICs or auxkernels on timestep_begin, where the variables won't change in `updateMesh` during residual evaluations. In the test I modified here, screen output from the user object only appears twice, but the `updateMesh` function is called 18 times.

Reply:
I guess that your sweeper is expensive to reinit?

Reply:
It is not light, i.e. going through a lot of code, but possibly not very expensive (did not tally the time before). It will probably be quite concerning if we have to call its init over and over again.